### PR TITLE
host_numa_info: add case

### PR DESCRIPTION
**Pull request by dzhengfy**
Case ID:  xxx-297812
Test scenarios:  Verify numa related info could be achieved by allocating hugepages using supported hugepage size,
check numa information by virsh commands, like capabilities, freepages...

Signed-off-by: Dan Zheng <dzheng@redhat.com>

### CODE REVIEW

**Comment by dzhengfy** [comment]
(1/1) type_specific.io-github-autotest-libvirt.host_numa.numa_info.default: PASS (45.78 s)
RESULTS    : PASS 1 | ERROR 0 | FAIL 0 | SKIP 0 | WARN 0 | INTERRUPT 0 | CANCEL 0
JOB HTML   : /var/lib/avocado/job-results/job-2023-09-17T06.53-c5718c7/results.html
JOB TIME   : 49.05 s

libvirt-9.5.0

**Comment by dzhengfy** [comment]
Depend on 
- https://github.com/avocado-framework/avocado-vt/pull/3760
- https://github.com/avocado-framework/avocado-vt/pull/3753
- https://github.com/avocado-framework/avocado-vt/pull/3757
- https://github.com/avocado-framework/avocado-vt/pull/3752
- https://github.com/avocado-framework/avocado-vt/pull/3758

**Comment by dzhengfy** [comment]
Result on RHEL 8.x

 (1/1) type_specific.io-github-autotest-libvirt.host_numa.numa_info.default: WARN: Test passed but there were warnings during execution. Check the log for details. (4.74 s)
RESULTS    : PASS 0 | ERROR 0 | FAIL 0 | SKIP 0 | WARN 1 | INTERRUPT 0 | CANCEL 0
JOB HTML   : /var/lib/avocado/job-results/job-2023-10-07T01.40-796ba07/results.html
JOB TIME   : 5.09 s

libvirt-8.0.0

**Comment by dzhengfy** [comment]
@hs0210 Could you help check if it can pass on aarch64? Thanks

File: libvirt/tests/cfg/numa/host_numa/host_numa_info.cfg [hunk 1]
```diff
@@ -0,0 +1,10 @@
+- host_numa.numa_info:
+    type = host_numa_info
+    take_regular_screendumps = no
+    start_vm = "no"
+    no s390-virtio
+    variants:
+        - default:
+            allocate_dict = {2048: 200, 1048576: 2}
+            aarch64:
+                allocate_dict = {2048: 200, 524288: 4, 16777216: 2}
```
Comment (hs0210) [inline libvirt/tests/cfg/numa/host_numa/host_numa_info.cfg:10]: It is enough to test 2M and 512M on kernel 64K machine. 16G is too large to allocate on most machines.

From xxxx-9.3 aarch64 support two serials huge pagesize according to the kernel pagesize.
For xxxx-9 kernel 4K, the supported huge pagesizes are 64K, 2M, 32M, 1G and default to 2M.
For xxxx-9 kernel 64K, the supported huge pagesizes are 2M, 512M, 16G and default to 512M, same with xxxx-8 aarch64.

Please adapt the test cases according to the above info.

File: libvirt/tests/src/numa/host_numa/host_numa_info.py [hunk 2]
```diff
@@ -0,0 +1,378 @@
...
+    test_obj.params['default_page_size'] = default_page_size
+    test_obj.params['supported_page_size_list'] = supported_page_size_list
+
+    allocate_memory_on_host_nodes(test_obj)
+    verify_node_mem_by_capabilities(test_obj)
+    verify_node_mem_by_freepages_all(test_obj)
+    verify_node_mem_by_freepages(test_obj)
+    verify_node_mem_by_freepages_various_unit(test_obj)
+    verify_node_distance_by_capabilities(test_obj)
+    verify_node_cpus_by_capabilities(test_obj)
```
Comment (hs0210) [inline libvirt/tests/src/numa/host_numa/host_numa_info.py:348]: Please skip this verification for aach64 as aarch64 does not have dies.

**Comment by dzhengfy** [comment]
(1/1) type_specific.io-github-autotest-libvirt.host_numa.numa_info.default: PASS (8.69 s)
on arm 8.0.0

File: libvirt/tests/src/numa/host_numa/host_numa_info.py (hunk 2 above)
Comment (dzhengfy) [reply libvirt/tests/src/numa/host_numa/host_numa_info.py]: Updated

File: libvirt/tests/cfg/numa/host_numa/host_numa_info.cfg (hunk 1 above)
Comment (dzhengfy) [reply libvirt/tests/cfg/numa/host_numa/host_numa_info.cfg]: As we discussed, 4k will be enabled later. 16G has a problem now. I removed it.

**Review by hs0210 — APPROVED** [review]
LGTM

File: libvirt/tests/cfg/numa/host_numa/host_numa_info.cfg [hunk 3]
```diff
@@ -0,0 +1,12 @@
...
+    take_regular_screendumps = no
+    start_vm = "no"
+    no s390-virtio
+    variants:
+        - default:
+            allocate_dict = {2048: 200, 1048576: 2}
+            aarch64:
+                allocate_dict = {2048: 200, 524288: 4}
+
+
```
Comment (chloerh) [inline libvirt/tests/cfg/numa/host_numa/host_numa_info.cfg:12]: Too may blank lines?

Comment (dzhengfy) [reply libvirt/tests/cfg/numa/host_numa/host_numa_info.cfg]: updated

**Pull request merged**